react/features/settings/actions.web.ts

```diff
@@ -28,15 +31,33 @@ import {
     getShortcutsTabProps
 } from './functions.web';
 
+
 /**
  * Opens {@code LogoutDialog}.
  *
- * @param {Function} onLogout - The event in {@code LogoutDialog} that should be
- *  enabled on click.
+ * @param {string} logoutUrl - The url for logging out.
  * @returns {Function}
  */
-export function openLogoutDialog(onLogout: Function) {
-    return openDialog(LogoutDialog, { onLogout });
+export function openLogoutDialog(logoutUrl: string) {
+    return (dispatch: IStore['dispatch'], getState: IStore['getState']) => {
+        const state = getState();
+        const { conference } = state['features/base/conference'];
+
+        dispatch(openDialog(LogoutDialog, {
+            onLogout() {
+                if (browser.isElectron()) {
+                    const url = appendURLHashParam(logoutUrl, 'electron', 'true');
+
+                    window.open(url, '_blank');
+                    dispatch(hangup(true));
+                } else {
+                    window.location.href = logoutUrl;
+                }
+            }
+        }));
+
+        conference?.room.moderator.logout(() => dispatch(hangup(true)));
```

Comment:
Doesn't this need to be inside the onLogout function? Also it would likely need to run before the redirects.

Reply:
I moved inside onLogout function.